lib/pbench/agent/tool_meister.py

```diff
@@ -1316,121 +1316,120 @@ def _send_directory(self, directory, uri, ctx):
         target_dir = directory.name
         parent_dir = directory.parent
         tar_file = parent_dir / f"{target_dir}.tar.xz"
-        o_file = parent_dir / f"{target_dir}.tar.out"
-        e_file = parent_dir / f"{target_dir}.tar.err"
+        tar_args = [
+            self.tar_path,
+            "--create",
+            "--xz",
+            "--force-local",
+            f"--file={tar_file}",
+            target_dir,
+        ]
         try:
             # Invoke tar directly for efficiency.
-            with o_file.open("w") as ofp, e_file.open("w") as efp:
-                cp = subprocess.run(
-                    [
-                        self.tar_path,
-                        "--create",
-                        "--xz",
-                        "--force-local",
-                        f"--file={tar_file}",
-                        target_dir,
-                    ],
-                    cwd=parent_dir,
-                    stdin=None,
-                    stdout=ofp,
-                    stderr=efp,
-                )
+            cp = subprocess.run(
+                tar_args,
+                cwd=parent_dir,
+                stdin=None,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
         except Exception:
             self.logger.exception("Failed to create tar ball '%s'", tar_file)
             failures += 1
         else:
             try:
                 if cp.returncode != 0:
-                    self.logger.error(
-                        "Failed to create tar ball; return code: %d", cp.returncode,
+                    self.logger.warning(
+                        "Failed to create tar ball first time; Error: %s, re-trying with --warning=none",
+                        str(cp.stdout),
```

Comment:
Perhaps consider using `cp.stdout.decode("utf-8")`.

Also, perhaps re-wording to consider?  `Tar ball creation failed with '%s', re-trying ignoring all warnings using --warning=none`

Reply:
I wondered if we should include `cp.returncode`? Looking at `tar` and `xz` I'm not sure these are particularly valuable (basically it'll be 1 or 2, and interpreting the actual meaning will be down to analyzing the `stderr` output anyway). Still... perhaps we should?

Reply:
Yep adding returncode doesn't hurt, added it in the latest commit.